servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/CatchAllLoadBalancerObserver.java

```diff
@@ -0,0 +1,135 @@
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+final class CatchAllLoadBalancerObserver implements LoadBalancerObserver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CatchAllLoadBalancerObserver.class);
+
+    private final LoadBalancerObserver delegate;
+
+    private CatchAllLoadBalancerObserver(LoadBalancerObserver delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public HostObserver hostObserver(Object resolvedAddress) {
+        try {
+            return new CatchAllHostObserver(delegate.hostObserver(resolvedAddress));
+        } catch (Throwable ex) {
+            LOGGER.warn("Unexpected exception from {} while getting a HostObserver", delegate, ex);
+            return NoopLoadBalancerObserver.NoopHostObserver.INSTANCE;
+        }
+    }
+
+    @Override
+    public void onServiceDiscoveryEvent(Collection<? extends ServiceDiscovererEvent<?>> events) {
+        safeReport(() -> delegate.onServiceDiscoveryEvent(events), "onServiceDiscoveryEvent");
+    }
+
+    @Override
+    public void onHostsUpdate(Collection<? extends Host> oldHosts, Collection<? extends Host> newHosts) {
+        safeReport(() -> delegate.onHostsUpdate(oldHosts, newHosts), "onHostsUpdate");
+    }
+
+    @Override
+    public void onNoAvailableHostException(NoAvailableHostException exception) {
+        safeReport(() -> delegate.onNoAvailableHostException(exception), "onNoAvailableHostException");
+    }
+
+    @Override
+    public void onNoActiveHostException(Collection<? extends Host> hosts, NoActiveHostException exception) {
+        safeReport(() -> delegate.onNoActiveHostException(hosts, exception), "onNoActiveHostException");
+    }
+
+    private static final class CatchAllHostObserver implements HostObserver {
+
+        private final HostObserver delegate;
+
+        CatchAllHostObserver(HostObserver delegate) {
+            this.delegate = requireNonNull(delegate, "delegate");
+        }
+
+        @Override
+        public void onHostMarkedExpired(int connectionCount) {
+            safeReport(() -> delegate.onHostMarkedExpired(connectionCount), "onHostMarkedExpired");
+        }
+
+        @Override
+        public void onActiveHostRemoved(int connectionCount) {
+            safeReport(() -> delegate.onActiveHostRemoved(connectionCount), "onActiveHostRemoved");
+        }
+
+        @Override
+        public void onExpiredHostRevived(int connectionCount) {
+            safeReport(() -> delegate.onExpiredHostRevived(connectionCount), "onExpiredHostRevived");
+        }
+
+        @Override
+        public void onExpiredHostRemoved(int connectionCount) {
+            safeReport(() -> delegate.onExpiredHostRemoved(connectionCount), "onExpiredHostRemoved");
+        }
+
+        @Override
+        public void onHostMarkedUnhealthy(@Nullable Throwable cause) {
+            safeReport(() -> delegate.onHostMarkedUnhealthy(cause), "onHostMarkedUnhealthy");
```

Comment:
What are your thoughts on runtime allocation of lambdas with a captured state?
The `CatchAllTransportObserver` was created long time ago I didn't really considered that there. Later, in `AbstractLifecycleObserverHttpFilter`, I tried to avoid capturing lambdas to minimize unnecessary heap allocations. It results in little bit more code though. 

Here we have use-cases with zero, one, two arguments and with primitive arguments. Don't wanna go the route with multiple `safeReport` overloads for different functions, which will also cause auto-boxing. But maybe we can just do `try-catch` in every method and only creating one `report` method to unify `LOGGER.warn` call?

Reply:
I understand your concern but this is much more readable. Let's consult the profiler to see if the lambdas end up stack allocated and if it's not then let's inline it.